src/api/search/src/bin/validation.js

```diff
@@ -31,8 +31,60 @@ const queryValidationRules = [
     .bail(),
 ];
 
-const validateQuery = (rules) => {
+/**
+ * Advanced search is more flexible, only needs at least ONE field, but can run without any too.
+ * Date formats must be YYYY-MM-DD
+ */
+const advancedQueryValidationRules = [
+  oneOf([
+    check('postText')
+      .exists({ checkFalsy: true })
+      .withMessage('post should not be empty')
+      .bail()
+      .isLength({ max: 256, min: 1 })
+      .withMessage('post should be between 1 to 256 characters')
+      .bail(),
+    check('authorText')
+      .exists({ checkFalsy: true })
+      .withMessage('author should exist')
+      .bail()
+      .isLength({ max: 100, min: 2 })
+      .withMessage('invalid author value')
+      .bail(),
+    check('titleText')
+      .exists({ checkFalsy: true })
+      .withMessage('title should exist')
+      .bail()
+      .isLength({ max: 100, min: 2 })
+      .withMessage('invalid title value')
+      .bail(),
+  ]),
+  check('dateEnd').optional().isISO8601().withMessage('invalid date format').bail(),
+
+  check('dateStart').optional().isISO8601().withMessage('invalid date format').bail(),
+  check('perPage')
+    .optional()
+    .isInt({ min: 1, max: 10 })
+    .withMessage('perPage should be empty or a number between 1 to 10')
+    .bail(),
+
+  check('page')
+    .optional()
+    .isInt({ min: 0, max: 999 })
+    .withMessage('page should be empty or a number between 0 to 999')
+    .bail(),
+];
+
+/**
+ * Validates query by passing rules. The rules are different based on the pathname
+ * of the request. If the pathname is '/' it is the basic route.
+ * Otherwise, if '/advanced/' it is the advanced search
+ */
+const validateQuery = () => {
   return async (req, res, next) => {
+    const rules =
+      req._parsedUrl.pathname === '/' ? queryValidationRules : advancedQueryValidationRules;
```

Comment:
This is failing eslint

Reply:
Can we change the eslint rules? or should I try another method?

Reply:
Whenever possible, we try to listen to eslint's advice.  Here it's advising you not to rely on an internal API, which could change.  I would use the public API, `originalUrl`, see https://expressjs.com/en/api.html#req.originalUrl